cmd/cstor-pool-mgmt/controller/backup-controller/handler.go

```diff
@@ -0,0 +1,237 @@
+	if err != nil {
+		runtime.HandleError(fmt.Errorf("invalid resource key: %s", key))
+		return nil, nil
+	}
+
+	bkp, err := c.clientset.OpenebsV1alpha1().BackupCStors(ns).Get(name, metav1.GetOptions{})
+	if err != nil {
+		if errors.IsNotFound(err) {
+			runtime.HandleError(fmt.Errorf("bkp '%s' in work queue no longer exists", key))
+			return nil, nil
+		}
+		return nil, err
+	}
+	return bkp, nil
+}
+
+// IsPendingStatus is to check if the backup is in a pending state.
+func IsPendingStatus(bkp *apis.BackupCStor) bool {
+	if string(bkp.Status) == string(apis.BKPCStorStatusPending) {
+		return true
+	}
+	return false
+}
+
+// IsInProgressStatus is to check if the backup is in in-progress state.
+func IsInProgressStatus(bkp *apis.BackupCStor) bool {
+	if string(bkp.Status) == string(apis.BKPCStorStatusInProgress) {
+		return true
+	}
+	return false
+}
+
+// IsInitStatus is to check if the backup is in init state.
+func IsInitStatus(bkp *apis.BackupCStor) bool {
+	if string(bkp.Status) == string(apis.BKPCStorStatusInit) {
+		return true
+	}
+	return false
+}
+
+// IsDoneStatus is to check if the backup is completed or not
+func IsDoneStatus(bkp *apis.BackupCStor) bool {
+	if string(bkp.Status) == string(apis.BKPCStorStatusDone) {
+		return true
+	}
+	return false
+}
+
+// IsFailedStatus is to check if the backup is failed or not
+func IsFailedStatus(bkp *apis.BackupCStor) bool {
+	if string(bkp.Status) == string(apis.BKPCStorStatusFailed) {
+		return true
+	}
+	return false
+}
+
+// IsRightCStorPoolMgmt is to check if the backup request is for this cstor-pool or not.
+func IsRightCStorPoolMgmt(bkp *apis.BackupCStor) bool {
+	if os.Getenv(string(common.OpenEBSIOCStorID)) == bkp.ObjectMeta.Labels["cstorpool.openebs.io/uid"] {
+		return true
+	}
+	return false
+}
+
+// IsDestroyEvent is to check if the call is for backup destroy.
+func IsDestroyEvent(bkp *apis.BackupCStor) bool {
+	if bkp.ObjectMeta.DeletionTimestamp != nil {
+		return true
+	}
+	return false
+}
+
+// IsOnlyStatusChange is to check the only status change of BackupCStor object.
+func IsOnlyStatusChange(oldbkp, newbkp *apis.BackupCStor) bool {
+	if reflect.DeepEqual(oldbkp.Spec, newbkp.Spec) &&
+		!reflect.DeepEqual(oldbkp.Status, newbkp.Status) {
+		return true
+	}
+	return false
+}
+
+func (c *BackupController) updateBackupCStorLast(bkp *apis.BackupCStor) error {
+	lastbkpname := bkp.Spec.BackupName + "-" + bkp.Spec.VolumeName
+	bkplast, err := c.clientset.OpenebsV1alpha1().BackupCStorLasts(bkp.Namespace).Get(lastbkpname, v1.GetOptions{})
+	if err != nil {
+		glog.Errorf("Failed to get last completed backup for %s vol:%v", bkp.Spec.BackupName, bkp.Spec.VolumeName)
+		return nil
+	}
+
+	bkplast.Spec.SnapName = bkplast.Spec.PrevSnapName
+	bkplast.Spec.PrevSnapName = bkp.Spec.SnapName
+	_, err = c.clientset.OpenebsV1alpha1().BackupCStorLasts(bkp.Namespace).Update(bkplast)
+	if err != nil {
+		glog.Errorf("Failed to update lastbackup for %s", bkplast.Name)
```

Comment:
```go
err =  errors.Wrapf(err, "failed to update backupcstorlast: current backup {%s}: last backup {%s}", bkp, bkplast)
glog.Errorf(err)
return err
```

Reply:
We can log only at a single place i.e. at the source of these calls.

Reply:
ok